### PR TITLE
Add DeiT models

### PR DESCRIPTION
**Pull request by sayakpaul**
DeiT: [Training data-efficient image transformers & distillation through attention](https://arxiv.org/abs/2012.12877)

### CODE REVIEW

**Comment by sayakpaul** [comment]
@WGierke have I placed the documentation in the wrong location? What am I missing out here?

File: assets/docs/sayakpaul/collections/deit/1.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+|  6 | deit_base_distilled_patch16_224  |        83.326 |        96.496 |               83.4 |               96.5 |
+|  7 | deit_base_distilled_patch16_384  |        85.238 |        97.172 |               85.2 |               97.2 |
+
+Note that the top-1 and top-5 accuracies are reported on the ImageNet-1k validation set. 
+[This directory](https://github.com/sayakpaul/deit-tf/tree/main/i1k_eval) provides details
+on how these numbers were generated.
+
+### Image classifiers
+
+* [deit_tiny_patch16_224](https://tfhub.dev/sayakpaul/deit_tiny_1k_224/1)
```
Comment (WGierke) [inline assets/docs/sayakpaul/collections/deit/1.md:40]: deit_tiny_1k_224?

File: assets/docs/sayakpaul/collections/deit/1.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+* [deit_small_distilled_patch16_224](https://tfhub.dev/sayakpaul/deit_small_distilled_1k_224/1)
+* [deit_base_patch16_224](https://tfhub.dev/sayakpaul/deit_base_1k_224/1)
+* [deit_base_distilled_patch16_224](https://tfhub.dev/sayakpaul/deit_base_distilled_1k_224/1)
+* [deit_base_patch16_384](https://tfhub.dev/sayakpaul/deit_base_1k_384/1)
+* [deit_base_distilled_patch16_384](https://tfhub.dev/sayakpaul/deit_base_distilled_1k_384/1)
+
+
+### Feature extractors
+
+* [deit_tiny_patch16_224_fe](https://tfhub.dev/sayakpaul/deit_tiny_1k_224_fe/1)
```
Comment (WGierke) [inline assets/docs/sayakpaul/collections/deit/1.md:52]: deit_tiny_1k_224_fe?

File: assets/docs/sayakpaul/collections/deit/1.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+Models included in this collection have two variants: (1) off-the-shelf inference for image
+classification, (2) fine-tuning on downstream tasks. These models are accompanied by
+Colab Notebooks for demonstration purposes. 
+
+The table below provides a performance summary:
+
+|    | **model_name**                       |   **top1_acc(%)** |   **top5_acc(%)** |   **orig_top1_acc(%)** |   **orig_top5_acc(%)** |
+|---:|:---------------------------------|--------------:|--------------:|-------------------:|-------------------:|
+|  0 | deit_tiny_patch16_224            |        72.136 |        91.128 |               72.2 |               91.1 |
```
Comment (WGierke) [inline assets/docs/sayakpaul/collections/deit/1.md:25]: deit_tiny_1k_224?

**Review by WGierke — CHANGES_REQUESTED** [review]
Looks great! I added some minor comments

File: assets/docs/sayakpaul/collections/deit/1.md (hunk 3 above)
Comment (sayakpaul) [reply assets/docs/sayakpaul/collections/deit/1.md:25]: No the identifier is correct.

**Review by WGierke — APPROVED** [review]

**Comment by sayakpaul** [comment]
@WGierke the model identifiers were not matching previously and it didn't sit right with me. The latest changes (https://github.com/tensorflow/tfhub.dev/pull/118/commits/609890c8f894aa1c114a3847c58793556dfd9f61) fix that.

**Review by WGierke — APPROVED** [review]

**Pull request merged**